pkg/download/grailfiltersegment/download_test.go

```diff
@@ -0,0 +1,166 @@
+/*
+ * @license
+ * Copyright 2024 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package grailfiltersegment_test
+
+import (
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/dynatrace/dynatrace-configuration-as-code-core/api/rest"
+	coreLib "github.com/dynatrace/dynatrace-configuration-as-code-core/clients/grailfiltersegments"
+	"github.com/dynatrace/dynatrace-configuration-as-code-core/testutils"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/featureflags"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/config"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/download/grailfiltersegment"
+)
+
+func TestDownloader_Download(t *testing.T) {
```

Comment:
I'd like to discuss with the team here what we want to do. In our last sync about testing, I think we agreed that we should use fakes in these cases, and only testservers if we interact directly with the API.

Reply:
As discussed in a meeting, the test has been rewritten.